samza-core/src/test/java/org/apache/samza/zk/TestZkProcessorLatch.java

```diff
@@ -39,256 +28,193 @@
+          zkUtils.close();
         }
       });
 
-    Future f2 = pool.submit(
-      () -> {
-        Latch latch = coordinationUtils.getLatch(latchSize, latchId);
-        latch.countDown();
-        try {
-          latch.await(100000, TimeUnit.MILLISECONDS);
-        } catch (TimeoutException e) {
-          Assert.fail("await timed out. " + e.getLocalizedMessage());
-        }
-      });
+    Future f2 = pool.submit(getParticipantRunnable(latchSize, latchId, "participant2"));
 
     try {
-      f1.get(30000, TimeUnit.MILLISECONDS);
-      f2.get(30000, TimeUnit.MILLISECONDS);
+      f1.get(30, TimeUnit.SECONDS);
+      f2.get(30, TimeUnit.SECONDS);
     } catch (Exception e) {
       Assert.fail("failed to get future." + e.getLocalizedMessage());
+    } finally {
+      pool.shutdownNow();
     }
-    pool.shutdownNow();
-  }
-
-  @Test
-  public void testNSizeLatch() {
-    System.out.println("Started N");
-    String latchId = "l1";
-    int latchSize = 3;
-
-    ExecutorService pool = Executors.newFixedThreadPool(3);
-    Future f1 = pool.submit(
-      () -> {
-        Latch latch = coordinationUtils.getLatch(latchSize, latchId);
-        latch.countDown();
-        try {
-          latch.await(100000, TimeUnit.MILLISECONDS);
-        } catch (TimeoutException e) {
-          Assert.fail("await timed out " + e.getLocalizedMessage());
-        }
-      });
-    Future f2 = pool.submit(
-      () -> {
-        Latch latch = coordinationUtils.getLatch(latchSize, latchId);
-        latch.countDown();
-        try {
-          latch.await(100000, TimeUnit.MILLISECONDS);
-        } catch (TimeoutException e) {
-          Assert.fail("await timed out. " + e.getLocalizedMessage());
-        }
-      });
-    Future f3 = pool.submit(
-      () -> {
-        Latch latch = coordinationUtils.getLatch(latchSize, latchId);
-        latch.countDown();
-        try {
-          latch.await(100000, TimeUnit.MILLISECONDS);
-        } catch (TimeoutException e) {
-          Assert.fail("await timed out. " + e.getLocalizedMessage());
-        }
-      });
-
     try {
-      f1.get(300, TimeUnit.MILLISECONDS);
-      f2.get(300, TimeUnit.MILLISECONDS);
-      f3.get(300, TimeUnit.MILLISECONDS);
+      List<String> latchParticipants =
+          testZkUtils.getZkClient().getChildren(
+              String.format("%s/%s_%s", KEY_BUILDER.getRootPath(), ZkProcessorLatch.LATCH_PATH, latchId));
+      Assert.assertNotNull(latchParticipants);
+      Assert.assertEquals(1, latchParticipants.size());
+      Assert.assertEquals("0000000000", latchParticipants.get(0));
     } catch (Exception e) {
-      Assert.fail("failed to get future. " + e.getLocalizedMessage());
+      Assert.fail("Failed to read the latch status from ZK directly" + e.getLocalizedMessage());
     }
   }
 
   @Test
-  public void testLatchExpires() {
-    System.out.println("Started expiring");
-    String latchId = "l4";
-
-    int latchSize = 3;
+  public void testLatchSizeN() {
```

Comment:
I am confused of the test names now. :/ testLatchSizeN vs testLatchSizeOneWithNParticipants, which one is more accurate here?